pandora-client-web/src/components/settings/securitySettings.tsx

```diff
@@ -0,0 +1,273 @@
+import type { AuthToken } from '../../networking/directoryConnector';
+import { useCurrentTime } from '../../common/useCurrentTime';
+import { useKeyDownEvent } from '../../common/useKeyDownEvent';
+import { Row } from '../common/container/container';
+
+export function SecuritySettings(): ReactElement | null {
+	const account = useCurrentAccount();
+
+	if (!account)
+		return <>Not logged in</>;
+
+	return (
+		<>
+			<ConnectedClients />
+			<PasswordChange account={ account } />
+		</>
+	);
+}
+
+type AccountConnectedClients = IClientDirectoryNormalResult['queryConnections']['connections'];
+
+function ConnectedClients(): ReactElement {
+	const directoryConnector = useDirectoryConnector();
+	const [connections, setConnections] = React.useState<AccountConnectedClients | null>(null);
+
+	const [load, processing] = useAsyncEvent(
+		async () => await directoryConnector.awaitResponse('queryConnections', {}),
+		(resp) => setConnections(resp.connections),
+	);
+
+	const onChange = React.useCallback((open: boolean) => {
+		if (open && connections === null) {
+			void load();
+		}
+	}, [connections, load]);
+
+	return (
+		<FieldsetToggle legend='Connected Clients' open={ false } onChange={ onChange }>
+			<div>
+				<ConnectedClientsList connections={ connections } />
+				<Button onClick={ load } disabled={ processing }>Refresh</Button>
+			</div>
+		</FieldsetToggle>
+	);
+}
+
+function ConnectedClientsList({ connections }: { connections: AccountConnectedClients | null; }): ReactElement {
+	if (connections === null)
+		return <>Loading...</>;
+
+	return (
+		<table>
+			<thead>
+				<tr>
+					<th>Id</th>
+					<th>Connections</th>
+					<th>Characters</th>
+					<th>Actions</th>
+				</tr>
+			</thead>
+			<tbody>
+				{
+					connections.map((c) => <ConnectedClientConnection key={ c.loginTokenId } connection={ c } />)
+				}
+			</tbody>
+		</table>
+	);
+}
+
+function ConnectedClientConnection({ connection }: { connection: AccountConnectedClients[number]; }): ReactElement {
+	const directoryConnector = useDirectoryConnector();
+	const authToken = useObservable(directoryConnector.authToken);
+	const confirm = useConfirmDialog();
+
+	const { loginTokenId, connectionCount, connectedCharacters } = connection;
+
+	const disconnect = React.useCallback(() => {
+		void confirm('Are you sure you want to disconnect this client?').then((confirmed) => {
+			if (confirmed) {
+				directoryConnector.sendMessage('logout', { type: 'selected', accountTokenId: loginTokenId });
+			}
+		});
+	}, [confirm, directoryConnector, loginTokenId]);
+
+	const isCurrent = authToken?.value.startsWith(loginTokenId);
+
+	return (
+		<tr className={ isCurrent ? 'current-connection' : undefined }>
+			<td>{ loginTokenId }</td>
+			<td>{ connectionCount }</td>
+			<td>{ connectedCharacters.map(({ id, name }) => `${name} (${id})`).join(', ') }</td>
+			<td>
+				{
+					isCurrent ? (
+						<>
+							<Button className='slim' onClick={ disconnect }>Logout</Button>
```

Comment:
I'm not fully sure how I feel about the logout button for current session. I feel having only one in the top right might be simpler, this button not adding any value besides having multiple ways to do the same thing.